rPackages.deldir fails to build on Darwin

### Steps To Reproduce

1. Build *deldir* (`nix-shell -p pkgs.rPackages.deldir`).

### Build log

Tail ([full log](https://gist.github.com/Aehmlo/cfad6364c3eb68788f158f06fc5aac1a)):
```
/nix/store/j8w19vhk1yazy96zm22apalqm1jvvlrg-clang-wrapper-11.1.0/bin/cc -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/nix/store/304fs644xapqnaz5wzh805inlfi1gmj2-R-4.2.1/lib/R/lib -L/nix/store/8qr4a99758ck931xj312yzj032jlq8v3-libcxx-11.1.0/lib -o deldir.so acchk.o addpt.o adjchk.o binsrt.o circen.o cross.o crossutil.o delet.o delet1.o delout.o delseg.o dirout.o dirseg.o dldins.o init.o initad.o insrt.o insrt1.o intri.o locn.o master.o mnnd.o pred.o qtest.o qtest1.o stoke.o succ.o swap.o testeq.o triar.o trifnd.o -L/nix/store/60g92jw5cbmsv853ajdfbnip4qp3gnw5-gfortran-11.2.0-lib/lib -L/nix/store/pfglcjndzqjjljq2rj7bzcbawxvha1sv-gfortran-wrapper-11.2.0/bin -L/nix/store/wpxisl5716sn2kipigg19pk6wrgaa21l-gfortran-11.2.0/lib/gcc/aarch64-apple-darwin/11.2.0 -L/nix/store/wpxisl5716sn2kipigg19pk6wrgaa21l-gfortran-11.2.0/lib -lintl -liconv -lgfortran -lm -L/nix/store/304fs644xapqnaz5wzh805inlfi1gmj2-R-4.2.1/lib/R/lib -lR -lintl -Wl,-framework -Wl,CoreFoundation
ld: library not found for -liconv
clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [/nix/store/304fs644xapqnaz5wzh805inlfi1gmj2-R-4.2.1/lib/R/share/make/shlib.mk:10: deldir.so] Error 1
ERROR: compilation failed for package ‘deldir’
* removing ‘/nix/store/xl1r02dz8fiw4rc3gz1da9n34czypyl7-r-deldir-1.0-6/library/deldir’
```

### Additional context

I am actually trying to install `pkgs.rPackages.interp`, of which `deldir` appears to be a transitive dependency.

### Notify maintainers

@jbedo (best guess per `git blame`)

### Metadata

```console
[alex@mor306-4:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.6.0, macOS 12.6`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.11.0`
 - channels(alex): `""`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

--- a/pkgs/development/r-modules/default.nix
+++ b/pkgs/development/r-modules/default.nix
@@ -498,6 +498,7 @@ let
 
   packagesWithBuildInputs = {
     # sort -t '=' -k 2
+    deldir = lib.optionals stdenv.isDarwin [ pkgs.libiconv ];
     gam = lib.optionals stdenv.isDarwin [ pkgs.libiconv ];
     RcppArmadillo = lib.optionals stdenv.isDarwin [ pkgs.libiconv ];
     quantreg = lib.optionals stdenv.isDarwin [ pkgs.libiconv ];

--- a/pkgs/development/r-modules/default.nix
+++ b/pkgs/development/r-modules/default.nix
@@ -500,6 +500,7 @@ let
     # sort -t '=' -k 2
     deldir = lib.optionals stdenv.isDarwin [ pkgs.libiconv ];
     gam = lib.optionals stdenv.isDarwin [ pkgs.libiconv ];
+    interp = lib.optionals stdenv.isDarwin [ pkgs.libiconv ];
     RcppArmadillo = lib.optionals stdenv.isDarwin [ pkgs.libiconv ];
     quantreg = lib.optionals stdenv.isDarwin [ pkgs.libiconv ];
     rmutil = lib.optionals stdenv.isDarwin [ pkgs.libiconv ];

